src/browser/constructLayoutEngine.js

```diff
@@ -170,29 +172,41 @@ export function constructLayoutEngine({
         ? document.querySelector(resolvedRoutes.containerEl)
         : resolvedRoutes.containerEl;
 
+    const location = evt ? strToLocation(evt.detail.newUrl) : window.location;
+
     recurseRoutes({
-      location: window.location,
+      location,
       routes: resolvedRoutes.routes,
       parentContainer,
       shouldMount: true,
       pendingRemovals,
+      pendingMakeVisibles,
     });
   }
 
-  function handleRoutingEvent({ detail: { appsByNewStatus } }) {
+  function handleRoutingEvent(evt) {
     pendingRemovals.forEach((remove) => remove());
     pendingRemovals = [];
 
-    appsByNewStatus.NOT_MOUNTED.concat(appsByNewStatus.NOT_LOADED).forEach(
-      (name) => {
-        const applicationElement = document.getElementById(
-          applicationElementId(name)
-        );
-        if (applicationElement && applicationElement.isConnected) {
-          applicationElement.parentNode.removeChild(applicationElement);
+    if (evt) {
+      const {
+        detail: { appsByNewStatus },
+      } = evt;
+
+      appsByNewStatus.NOT_MOUNTED.concat(appsByNewStatus.NOT_LOADED).forEach(
+        (name) => {
+          const applicationElement = document.getElementById(
+            applicationElementId(name)
+          );
+          if (applicationElement && applicationElement.isConnected) {
+            applicationElement.parentNode.removeChild(applicationElement);
+          }
         }
-      }
-    );
+      );
+    }
+
+    pendingMakeVisibles.forEach((makeVisible) => makeVisible());
```

Comment:
This approach has the downside of not making any of the dom elements visible until the slowest application has finished mounting. I think it's okay, though

Reply:
Could that mess with animations for transitioning in/out?

Reply:
Yes transitions in would be affected. Not transitions out, though.